File: android-kotlin/app/src/main/java/com/mongodb/app/Task.kt
package com.mongodb.app

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId

open class Task(
        @PrimaryKey var _id: ObjectId = ObjectId(),
        var _partition: String = "",
        var isComplete: Boolean = false,
        var summary: String = ""
): RealmObject() {}

File: xamarin/realm-todo-dotnet/Models/Task.cs

using MongoDB.Bson;
using Realms;

namespace RealmTemplateApp.Models
{
    public class Task : RealmObject
    {
        [PrimaryKey]
        [Required]
        [MapTo("_id")]
        public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

        // :state-start: partition-based-sync
        [MapTo("_partition")]
        [Required]
        public string Partition { get; set; }
        // :state-end:
        // :state-uncomment-start: flexible-sync
        //[MapTo("owner_id")]
        //[Required]
        //public string OwnerId { get; set; }
        // :state-uncomment-end:flexible-sync

        [MapTo("summary")]
        [Required]
        public string Summary { get; set; }

        [MapTo("isComplete")]
        public bool IsComplete { get; set; }
    }
}
